### PR TITLE
chore: Property filter token groups internal feature

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Introducing property filter token groups as an internal feature first.

Depends on https://github.com/cloudscape-design/collection-hooks/pull/74

Rel: [qUFhApKfmWEg]

### How has this been tested?

* New unit tests
* Manual testing
* Bug bash
* Screenshot tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json
Comment (pan-kot) [inline package.json:26]: This change is needed ton install the correct version of collection hooks, I will remove it before merging.

The code can be merged once https://github.com/cloudscape-design/collection-hooks/pull/74 is released.

File: src/property-filter/utils.ts
Comment (pan-kot) [reply src/property-filter/utils.ts:123]: The utility is used also on Token type when converting groups to tokens if enableTokenGroups=false:

```
if (enableTokenGroups) {
  fireNonCancelableEvent(onChange, { tokens: [], operation: query.operation, tokenGroups: tokens });
} else {
  fireNonCancelableEvent(onChange, { tokens: tokenGroupToTokens<Token>(tokens), operation: query.operation });
}
```

**Review by gethinwebster — APPROVED** [review]

**Pull request merged**
